### PR TITLE
Add order_by, limit and offset in find query

### PR DESCRIPTION
**Pull request by TheHelias**
Add order_by, limit and offset as options in find query parameters
Updated urllib3 for fix vulnerabilities

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,17 +74,21 @@ Querying is done via a MongoDB-like `find()`:
...
     product_id: str
     stock: int
+    unit_value: int
 
 
 Product.find({"product_id": "abc-123"})
 Product.find({"stock": {">=": 3}})
 # or
 Product.find({"stock": {op.GTE: 3}})
+Product.find({"stock": {">=": 3}}, order_by=[('unit_value', Query.ASCENDING)], limit=2, offset=3)
```
Comment (lietu) [inline README.md:90]: `>= 3`, `limit=2, offset=3` seem rather poor examples
```suggestion
Product.find({"stock": {">=": 1}}, order_by=[('unit_value', Query.ASCENDING)], limit=25, offset=50)
```

File: firedantic/tests/tests_async/test_model.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
+from operator import attrgetter
 from uuid import uuid4
 
 import pytest
+from google.cloud.firestore import (  # just for the ASCENDING / DESCENDING constants
+    Query,
+)
```
Comment (lietu) [inline firedantic/tests/tests_async/test_model.py:7]: Why is this comment and poor format needed?

```suggestion
from google.cloud.firestore import Query
```

**Comment by lietu** [comment]
If our typing expects you to use `Union[Literal["ASCENDING"], Literal["DESCENDING"]]`, then why do we use `Query.ASCENDING` and `Query.DESCENDING` in tests and examples? What are the guarantees that `Query` properties do not change?

**Comment by antont** [comment]
> If our typing expects you to use `Union[Literal["ASCENDING"], Literal["DESCENDING"]]`, then why do we use `Query.ASCENDING` and `Query.DESCENDING` in tests and examples? What are the guarantees that `Query` properties do not change?

Fair point, would at least make sense to use them in the typing as well.

Usually I dislike having strings in the code, so I'd like to use readily defined constants, and there they are already defined. If those change in the Firebase SDK, I guess that would then inflict changes to Firedantic as well anyway. I wouldn't expect those to change though.

File: firedantic/tests/tests_async/test_model.py [hunk 3]
```diff
@@ -150,6 +154,105 @@ async def test_find_array_contains_any(configure_db, create_todolist):
...
+    )
+    for company_id, last_name in ids_and_lastnames:
+        await create_company(company_id=company_id, last_name=last_name)
+
+    to_find = ids_and_lastnames[2]
+    find_id, find_lastname = to_find
+    c = await Company.find({"company_id": find_id, "owner.last_name": find_lastname})
+    company = c[0]
+    assert company.company_id == find_id
+    assert company.owner.last_name == find_lastname
```
[the remark this answers is not in the document]
Comment (antont) [reply firedantic/tests/tests_async/test_model.py]: I figure it's a bit a matter of taste, do you want a single test that tests everything about find, or multiple that test different features of it.

There was already e.g. `test_find_not_in` so I was following the pattern.

File: firedantic/_async/model.py [hunk 4]
```diff
@@ -96,25 +96,40 @@ def get_document_id(self):
...
         """Returns a list of models from the database based on a filter.
 
         Example: `Company.find({"company_id": "1234567-8"})`.
         Example: `Product.find({"stock": {">=": 1}})`.
+        Example: `Product.find(order_by=[('unit_value', "ASCENDING"), ('stock', "DESCENDING")], limit=2)`.
+        Example: `Product.find({"stock": {">=": 3}}, order_by=[('unit_value', "ASCENDING")], limit=2, offset=3)`.
 
         :param filter_: The filter criteria.
         :return: List of found models.
         """
```
Comment (lietu) [inline firedantic/_async/model.py:119]: image

File: firedantic/_async/model.py [hunk 5]
```diff
@@ -96,25 +96,44 @@ def get_document_id(self):
...
         Example: `Company.find({"company_id": "1234567-8"})`.
         Example: `Product.find({"stock": {">=": 1}})`.
+        Example: `Product.find(order_by=[('unit_value', Query.ASCENDING), ('stock', Query.DESCENDING)], limit=2)`.
+        Example: `Product.find({"stock": {">=": 3}}, order_by=[('unit_value', Query.ASCENDING)], limit=2, offset=3)`.
 
-        :param filter_: The filter criteria.
+        :param filter_: The filter criterion.
+        :param order_by: The order by criterion to sort the results.
+        :param limit: The limit criterion to set maximum number of results to return.
+        :param offset: The offset criterion to skip the first n results.
```
Comment (lietu) [inline firedantic/_async/model.py:121]:
```suggestion
        :param filter_: The filter criteria
        :param order_by: List of columns and direction to order results by
        :param limit: Maximum results to return
        :param offset: Skip the first n results
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -206,7 +212,8 @@ this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Update README.md
 - Update .gitignore
 
-[unreleased]: https://github.com/ioxiocom/firedantic/compare/0.5.1...HEAD
+[unreleased]: https://github.com/ioxiocom/firedantic/compare/0.6.0...HEAD
+[0.6.0]: https://github.com/ioxiocom/firedantic/compare/0.5.0...0.6.0
```
Comment (joakimnordling) [inline CHANGELOG.md:216]: This is actually still wrong, should be from 0.5.1 to 0.6.0

File: README.md [hunk 7]
```diff
@@ -74,17 +74,21 @@ Querying is done via a MongoDB-like `find()`:
...
     stock: int
+    unit_value: int
 
 
 Product.find({"product_id": "abc-123"})
 Product.find({"stock": {">=": 3}})
 # or
 Product.find({"stock": {op.GTE: 3}})
+Product.find({"stock": {">=": 3}}, order_by=[('unit_value', Query.ASCENDING)], limit=25, offset=50)
+Product.find(order_by=[('unit_value', "ASCENDING"), ('stock', Query.DESCENDING)], limit=2)
```
[the remark this answers is not in the document]
Comment (TheHelias) [reply README.md:91]: Not on purpose

**Comment by joakimnordling** [comment]
The same set of changes were submitted as a second PR https://github.com/ioxiocom/firedantic/pull/63 that was merged. Closing this one.

**Pull request closed without merging**